flare/gp.py

```diff
@@ -24,7 +24,7 @@ def __init__(self, kernel: Callable,
                  energy_kernel: Callable = None,
                  opt_algorithm: str = 'L-BFGS-B',
                  maxiter=10, par=False,
-                 monitor=False):
+                 output=None):

```

Comment:
Here and in a few other places, we could typehint output as the `Output` object.

Reply:
Type hints will be eliminated eventually, as they don't play well with the Sphinx documentation. Might be better to do without them.

Reply:
Sad to hear that Sphinx can't handle them. That's a shame. At least docstrings can contain the same info.

Reply:
Actually it looks like Sphinx only really struggles with np.ndarray type hints. Here's an example from the docs:
![image](https://user-images.githubusercontent.com/37220876/64886278-c77c5080-d633-11e9-8e42-d8b5e30aa5c5.png)

If we can find a way to get Sphinx to format that better, type hints should be okay.